jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/main/java/org/jbpm/workbench/ks/integration/KieServerIntegration.java

```diff
@@ -245,26 +245,41 @@ public void onServerInstanceDisconnected(@Observes ServerInstanceDisconnected se
     public void onServerInstanceConnected(@Observes ServerInstanceConnected serverInstanceConnected) {
         ServerInstance serverInstance = serverInstanceConnected.getServerInstance();
 
-        serverTemplatesClients.computeIfPresent(serverInstance.getServerTemplateId(),
-                                                (serverTemplateId, clients) -> {
-                                                    clients.forEach((key, client) -> {
-                                                        // update regular clients
-                                                        updateOrBuildClient(client,
-                                                                            serverInstance);
-
-                                                        logger.debug("KieServerClient load balancer updated for server template {}",
-                                                                     serverTemplateId.equals(SERVER_TEMPLATE_KEY) ? serverInstance.getServerTemplateId() : serverTemplateId);
-                                                    });
-                                                    return clients;
-                                                });
+        ServerInstanceKey serverInstanceKey = serverInstancesById.computeIfAbsent(serverInstance.getServerInstanceId(),s -> {
+            logger.debug("Server {} connected!", serverInstance);
+
+            if (Boolean.parseBoolean(System.getProperty("org.kie.server.startup.detectOpenshiftLoadBalancer", "false"))) {
+                serverInstancesById.put(serverInstance.getServerInstanceId(),
+                        serverInstance);
+            }
```

Comment:
```suggestion
            if (Boolean.parseBoolean(System.getProperty("org.kie.server.startup.detectOpenshiftLoadBalancer", "false"))) {
                serverInstancesById.put(serverInstance.getServerInstanceId(),
                        serverInstance);
            }
```
if (Boolean.getBoolean("org.kie.server.startup.detectOpenshiftLoadBalancer")) {
return serverInstance;
}